ArduPlane/mode.h

```diff
@@ -0,0 +1,475 @@
+#pragma once
+
+#include <stdint.h>
+
+#include <GCS_MAVLink/GCS_MAVLink.h>
+#include <AP_Math/AP_Math.h>
+
+#include "defines.h"
+
+#define MODE_NEXT_HEADING_UNKNOWN   99999.0f    // used to indicate to set_desired_location method that next leg's heading is unknown
+
+// pre-define ModeRTL so Auto can appear higher in this file
+class ModeRTL;
+
+class Mode
+{
+public:
+
+    // Auto Pilot modes
+    // ----------------
+    enum Number {
+        MANUAL        = 0,
+        CIRCLE        = 1,
+        STABILIZE     = 2,
+        TRAINING      = 3,
+        ACRO          = 4,
+        FLY_BY_WIRE_A = 5,
+        FLY_BY_WIRE_B = 6,
+        CRUISE        = 7,
+        AUTOTUNE      = 8,
+        AUTO          = 10,
+        RTL           = 11,
+        LOITER        = 12,
+        AVOID_ADSB    = 14,
+        GUIDED        = 15,
+        INITIALISING  = 16,
+        QSTABILIZE    = 17,
+        QHOVER        = 18,
+        QLOITER       = 19,
+        QLAND         = 20,
+        QRTL          = 21
+    };
+
+    // Constructor
+    Mode();
+
+    // enter this mode, returns false if we failed to enter
```

Comment:
Plane doesn't let you fail to enter a mode, does it?  That would make the comment misleading.

Reply:
fixed